### DISCUSSION TITLE
Merging of data

### DISCUSSION THREAD

**Discussion by LennartMrt** [KL-Psychological-Methodology/ESMira#16] [Questions] [answered]
Currently, we're doing a study with twelve different groups because at an early version of ESMira there was no possibility of randomization. Each group consists of a few questionnaires (for example the DASS or PMH, see screenshot). Currently, we can only export one questionnaire per group at a time. We would like to have all questionnaires per group in just one file. Is there any possibility of exporting the data of every questionnaire that is in one group at once? Or any other possible solution for this problem. 

Thanks in advance for your help! 

![Bildschirmfoto 2024-07-19 um 00 58 43](https://github.com/user-attachments/assets/ca44ef89-c820-48a3-93d4-dbdbed89207d)

**Comment by JodliDev** [comment, marked as answer]
Unfortunately there is no option to merge mutliple questionnaires. Also there is also no "bulk download" option (which might be a good idea for the furture?).
But if you have ftp access to your server, you can download the csv files directly, which should make it a lot more conventient than downloading them one by one through the browser.
You can find the files on your server in: `[path/to/your/esmira/installation]/esmira_data/studies/[studyId]/responses/`

> You can find the **studyId** in the browser url when you have opened your study in the admin interface. For example: The screenshot you have posted would have an url similar to: `https://DOMAIN.XX/?admin#admin/allStudies:data/dataStatistics,id:1234/dataList`. So the studyId for that study would be `1234`

**Reply by LennartMrt** [reply]
Thanks for your answer Jodli! 

What exactly do you mean by file transfer protocol access? Is admin access similar to ftp access? If not, how can I get ftp access? Because I can see the data for each questionnaire through the second url-example you've posted which is seen in my screenshot but I have to download each questionnaire by hand. The problem is that we would have to do that for each of our 12 versions. That would result in 240 (12x20) different csv files. After that, we would have to merge these 240 csv files. 

Best regards
Lennart

**Reply by JodliDev** [reply]
FTP is a protocol that is used to access files on your server. Through it, one can also access the files that are saved by ESMira.
Whoever setup your server also has FTP access. You could ask them to send you the contents of the `esmira_data` folder and then get the questionnaire files from the data structure (see above).
You would still have 240 files that you would need to merge on your own . But the process of downloading would be faster.

Unfortunately there is no feature to merge files through ESMira.
